src/components/input/input.js

```diff
@@ -70,22 +78,25 @@ export const Input = ({
       <div className={controlClassNames.join(" ")}>
         <input
           className={inputClasses}
-          type={type}
+          type={typeState}
           name={name}
           disabled={disabled}
           defaultValue={defaultValue}
           placeholder={usePlaceholder ? label : ""}
           onChange={onChange}
           required={required}
-          min={type === "number" && min ? min : void 0}
-          max={type === "number" && max ? max : void 0}
-          step={type === "number" && step ? step : void 0}
+          min={typeState === "number" && min ? min : void 0}
+          max={typeState === "number" && max ? max : void 0}
+          step={typeState === "number" && step ? step : void 0}
           minLength={minLength}
           maxLength={maxLength}
           ref={inputRef}
           pattern={pattern}
           title={title}
         />
+        {type === "password" && (
```

Comment:
I mean here

Reply:
The reason for `type` being used here, not `typeState` is that with `typeState` the `showIcon` would disappear after the first time of showing password. 
Also, the `showIcon` should be available only for input type of password, obviously.

Reply:
> Also, the `showIcon` should be available only for input type of password, obviously.

Then the same comment can be made for `min`, `max` and `step`: they should only be available for number input.